### ISSUE TITLE
testing (histograms): Need to complete and improve test tooling for native histograms

### ISSUE THREAD

**Issue by beorn7** [prometheus/prometheus#11171]
Here is a probably incomplete list:
- [ ] `tsdb/block_test.go` suffers from the lack of Histogram support in the `CreateBlock` helper. See https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/tsdb/block_test.go#L502 and https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/tsdb/tsdbblockutil.go#L58 
- [ ] https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/tsdb/compact_test.go#L1356 
- [ ] https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/tsdb/db_test.go#L95 
- [ ] https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/web/api/v1/api_test.go#L2787 
- [ ] https://github.com/prometheus/prometheus/blob/7feea8aa894da5f16d41935538d2bf6ffb55050f/web/ui/react-app/src/pages/graph/DataTable.test.tsx#L147

**Comment by codesome** [comment]
I have updated the description with all the different things that we need unit tests for (some already exist). I will update the list if I find something that was missed.

**Comment by beorn7** [comment]
Yeah, no clue how to do the UI testing properly. The little bits of code I wrote for Native Histogram support were totally makeshift. @Nexucis any help would be very welcome.

**Comment by Nexucis** [comment]
I don't have much bandwidth currently, but I will try.
In the meantime, perhaps @juliusv would be more capable to help here.

**Comment by codesome** [comment]
Moving UI test discussion to https://github.com/prometheus/prometheus/issues/11423. Other planned tests are done!
